pkg/kube/secret/secret.go

```diff
@@ -109,3 +109,25 @@ func HasAllKeys(secret corev1.Secret, keys ...string) bool {
 	}
 	return true
 }
+
+// EnsureSecretWithKey makes sure the Secret with the given name has a key with the given value if the key is not already present.
+// if the key is present, it will return the existing value associated with this key.
+func EnsureSecretWithKey(secretGetUpdateCreateDeleter GetUpdateCreateDeleter, nsName types.NamespacedName, key, value string) (string, error) {
+	existingSecret, err0 := secretGetUpdateCreateDeleter.GetSecret(nsName)
```

Comment:
[nit] I think we generally tend to reuse the name `err` every time? is there specific reason for keeping them separated here?

Reply:
there was a shadowing issue where the err1 check was referring to err0 when they were both `err`